util/stmtsummary/statement_summary.go

```diff
@@ -41,16 +41,19 @@ type stmtSummaryByDigestKey struct {
 	// Same statements may appear in different schema, but they refer to different tables.
 	schemaName string
 	digest     string
+	// The digest of the previous statement.
+	prevDigest string
 	// `hash` is the hash value of this object.
 	hash []byte
 }
 
 // Hash implements SimpleLRUCache.Key.
 func (key *stmtSummaryByDigestKey) Hash() []byte {
 	if len(key.hash) == 0 {
-		key.hash = make([]byte, 0, len(key.schemaName)+len(key.digest))
+		key.hash = make([]byte, 0, len(key.schemaName)+len(key.digest)+len(key.prevDigest))
 		key.hash = append(key.hash, hack.Slice(key.digest)...)
 		key.hash = append(key.hash, hack.Slice(key.schemaName)...)
+		key.hash = append(key.hash, hack.Slice(key.prevDigest)...)
```

Comment:
The same SQL statement is executed after different statements, the final digest is different? Is it expected?

Reply:
Actually, only when current SQL is `commit` do we record `prevSQL`. Otherwise, `prevSQL` is empty. See `adaptor.go`.
If all commits are grouped in one record, we can't distinguish different transactions.

Reply:
@djshow832 OK, It's a convention of coding and it's better to add some comments to hint the maintainer don't break it.

Reply:
Addressed.